### ISSUE TITLE
Regarding multi GPU training

### ISSUE THREAD

**Issue by Yifehuang97** [aipixel/GPS-Gaussian#37]
I apologize for reaching out again, but I've encountered an error in the diff-gaussian-rasterization while using multiple GPUs(Vanilla Torch DDP). It appears that the rasterizer's device may not be correctly configured, resulting in an illegal memory access error on all ranks except rank 0.

Have you experienced this issue before? If so, could you share how you managed to resolve it?

Thank you for your time and assistance!

**Comment by ShunyuanZheng** [comment]
Sorry, I have not trained on multiple GPUs. Maybe you can find some solutions on other repos for multi-GPU training with diff-gaussian-rasterization.

**Comment by Yifehuang97** [comment]
Thanks for your reply! I've solved it by using torch lightning.

**Comment by Langwenchong** [comment]
@Yifehuang97 Hello, I would like to ask you about how you specifically implemented this. If it’s convenient for you, could you please share a sample of your code with me? (My email is 1422257646@qq.com). If it’s not convenient, could you kindly point me to the post or issue you referred to when solving this problem? This issue is very important to me, and I’m looking forward to your reply!

**Comment by Yifehuang97** [comment]
> [@Yifehuang97](https://github.com/Yifehuang97) Hello, I would like to ask you about how you specifically implemented this. If it’s convenient for you, could you please share a sample of your code with me? (My email is [1422257646@qq.com](mailto:1422257646@qq.com)). If it’s not convenient, could you kindly point me to the post or issue you referred to when solving this problem? This issue is very important to me, and I’m looking forward to your reply!

Hi Langwen, I believe the main issue is that back in 2024, the differentiable GS renderer didn't support multi-GPU setups. However, Torch Lightning seemed to handle it internally at that time. I’ll check the code I used previously and see if I can find a working example. Also maybe you can take a look at LGM: Large Multi-View Gaussian Model for High-Resolution 3D Content Creation.

Best,
Yifeng

**Comment by Langwenchong** [comment]
@Yifehuang97 Hello, I have now modified gps_gaussian to a multi-GPU version combined with DDP. I didn’t use PyTorch Lightning — instead, each GPU separately calls the render function to render its own NVS views for computing the loss. I found that it seems possible to parallelize this independent rendering directly, but after training for a while, the GPUs crash and the driver drops completely. I’d like to ask if you encountered a similar issue back then?
